### PR TITLE
Add EncoderParameter tests for System.Drawing.Common.Imaginig for #20711

### PR DESCRIPTION
**Pull request by norek**
Hi little chunk of tests for Imiaginig part.
In EncoderParameterTests Ive used AnyEncoder propertie to inform that the value of this parameter in a particular case is irrelevant. What do you think about that, good or not.
In our company we are using something like Autofixture for this case.

### CODE REVIEW

File: src/System.Drawing.Common/tests/Imaging/EncoderParameterTests.cs [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.Drawing.Imaging.Tests
+{
+    public class EncoderParameterTests
+    {
+        private static Encoder AnyEncoder = Encoder.ChrominanceTable;
```
Comment (hughbe) [inline src/System.Drawing.Common/tests/Imaging/EncoderParameterTests.cs]: Nit: `AnyEncoder` -> `s_anyEncoder`

File: src/System.Drawing.Common/tests/Imaging/EncoderParameterTests.cs [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+                yield return new object[] { new Action(() => new EncoderParameter(null, new short[] { })) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, new long[] { })) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, new int[] { }, new int[] { })) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, new long[] { }, new long[] { })) };
+            }
+        }
+
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        [MemberData(nameof(EncoderParameterCtor_NullEncoder_TestData))]
+        public void Ctor_NullEncoder_ThrowsNullReferenceException(Action encoderCtor)
```
Comment (hughbe) [inline src/System.Drawing.Common/tests/Imaging/EncoderParameterTests.cs]: This seems unecessary - I know in general we/I say "use theory", but I'm not sure it's the best place for it here.

I'd just do 

```cs
public void Ctor_NullEncoder_ThrowsNullReferenceException(Action encoderCtor)
{
    Assert.Throws<NRE>(...):
    Assert.Throws<NRE>(...):
    Assert.Throws<NRE>(...):
    //...
}
``

File: src/System.Drawing.Common/tests/Imaging/EncoderParameterTests.cs [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+        [InlineData(EncoderParameterValueType.ValueTypeAscii, int.MaxValue, typeof(AccessViolationException))]
+        [InlineData(EncoderParameterValueType.ValueTypeShort, (int.MaxValue / 2) + 1, typeof(OverflowException))]
+        [InlineData(EncoderParameterValueType.ValueTypeLong, (int.MaxValue / 4) + 1, typeof(OverflowException))]
+        [InlineData(EncoderParameterValueType.ValueTypeRational, (int.MaxValue / 8) + 1, typeof(OverflowException))]
+        [InlineData(EncoderParameterValueType.ValueTypeLongRange, (int.MaxValue / 8) + 1, typeof(OverflowException))]
+        [InlineData(EncoderParameterValueType.ValueTypeUndefined, int.MaxValue, typeof(AccessViolationException))]
+        [InlineData(EncoderParameterValueType.ValueTypeRationalRange, (int.MaxValue / 16) + 1, typeof(OverflowException))]
+        public void Ctor_Encoder_TooBigNumberOfValues_Type_Value_AccessViolationException(EncoderParameterValueType type, int numberOfValues, Type expected)
+        {
+            IntPtr anyValue = IntPtr.Zero;
```
Comment (hughbe) [inline src/System.Drawing.Common/tests/Imaging/EncoderParameterTests.cs]: Nit: if you're changing this file for PR feedback you might as well inline this value

File: src/System.Drawing.Common/tests/Imaging/EncoderParametersTests.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+namespace System.Drawing.Imaging.Tests
+{
+    public class EncoderParametersTests
+    {
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void Ctor_Default()
+        {
+            EncoderParameters ep = new EncoderParameters();
+            Assert.NotNull(ep.Param);
+            Assert.Equal(1, ep.Param.Length);
```
Comment (hughbe) [inline src/System.Drawing.Common/tests/Imaging/EncoderParametersTests.cs]: You can just do `Assert.Equal(new EncoderParameter[1], ep.Param)`

File: src/System.Drawing.Common/tests/Imaging/EncoderParametersTests.cs [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            Assert.True(ep.Param.IsFixedSize);
+        }
+
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        [InlineData(1)]
+        public void Ctor_Count_Default(int count)
+        {
+            EncoderParameters ep = new EncoderParameters(count);
+            Assert.NotNull(ep.Param);
+            Assert.Equal(count, ep.Param.Length);
```
Comment (hughbe) [inline src/System.Drawing.Common/tests/Imaging/EncoderParametersTests.cs]: You can just do `Assert.Equal(new EncoderParameter[count], ep.Param)`

File: src/System.Drawing.Common/tests/Imaging/EncoderParameterTests.cs (hunk 2 above)
Comment (norek) [inline src/System.Drawing.Common/tests/Imaging/EncoderParameterTests.cs]: I still have a lot to learn but it's exciting!

Comment (hughbe) [inline src/System.Drawing.Common/tests/Imaging/EncoderParameterTests.cs]: Keep it up :D

**Comment by mellinoe** [comment]
@norek I'll second @hughbe 's suggestions. This looks very good overall. Thanks for the continued contribution!

**Comment by mellinoe** [comment]
@norek Do you have any time to do the final cleanup and rebasing for this PR?

**Comment by norek** [comment]
@mellinoe Hi, oh finally i got back my computer, sorry for this unexpected delay. Yes yes I have time, I want finish this PR since end of week

**Comment by mellinoe** [comment]
@norek Great, no rush! Just wanted to make sure this wasn't forgotten.

**Comment by norek** [comment]
@mellione unfortuntely i cant finish my work... On my new corp computer are blocked ports for gitgub! i dont know why and i cant push and pull anything. I must buy new private computer but it will be for about 3-4 weeknd. Can someone can finish this PR?

**Comment by mellinoe** [comment]
@norek I can clean this and finish it up sometime this week.

**Comment by mellinoe** [comment]
@norek Very sorry to hear about your work computer, by the way. Best of luck getting a new personal computer! We will always welcome your contributions 😄 .

**Comment by mellinoe** [comment]
I did final cleanup and rebased the branch.

**Comment by mellinoe** [comment]
@dotnet-bot Test UWP CoreCLR x64 Debug Build

**Comment by mellinoe** [comment]
The UWP failures are known issues (RemoteExecutor couldn't run, etc. etc.)

**Comment by mellinoe** [comment]
@norek Thanks for this contribution, it is much appreciated! Here's to hoping you can get a new personal computer soon 😄

**Pull request merged**